Library/src/Multiplayer/Components/TextSpaceComponent.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CSP/Multiplayer/Components/TextSpaceComponent.h"
+
+#include "Debug/Logging.h"
+#include "Memory/Memory.h"
+#include "Multiplayer/Script/ComponentBinding/TextSpaceComponentScriptInterface.h"
+
+
+namespace csp::multiplayer
+{
+
+TextSpaceComponent::TextSpaceComponent(SpaceEntity* Parent) : ComponentBase(ComponentType::Image, Parent)
+{
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Position)]			 = csp::common::Vector3::Zero();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Rotation)]			 = csp::common::Vector4 {0, 0, 0, 1};
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Scale)]				 = csp::common::Vector3::One();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Text)]				 = "";
+	Properties[static_cast<uint32_t>(TextPropertyKeys::TextColor)]			 = csp::common::Vector3::Zero();
```

Comment:
I believe the default text colour should be white as per the [PRD](https://magnopus.atlassian.net/wiki/spaces/OL/pages/2737963046/Text+Component#:~:text=Default%20will%20be%20black), however a vector3 zero is black I believe.

Reply:
Good catch!

Reply:
How are color channels represented in data in CSP? 0.0f-1.0f, or 0.0f-255.0f?

Reply:
0-1, where 0 is black and 1 is white.

Reply:
Addressed in latest 👍